src/tests/ftest/pool/destroy_tests.py

```diff
@@ -1,746 +1,631 @@
 
         :avocado: tags=pool,pooldestroy,destroybig
         """
-        try:
-            # write out a hostfile_servers and start the servers with it
-            self.hostlist_servers = self.params.get("test_machines6",
-                                                    '/run/hosts/')
-            hostfile_servers = write_host_file.write_host_file(
-                self.hostlist_servers, self.tmp)
-
-            self.agent_sessions = agent_utils.run_agent(self.basepath,
-                                                        self.hostlist_servers)
-            server_utils.run_server(hostfile_servers, self.server_group,
-                                    self.basepath)
-
-            # parameters used in pool create
-            createmode = self.params.get("mode", '/run/poolparams/createmode/')
-            createuid = self.params.get("uid", '/run/poolparams/createuid/')
-            creategid = self.params.get("gid", '/run/poolparams/creategid/')
-            createsetid = self.params.get("setname",
-                                          '/run/poolparams/createset/')
-            createsize = self.params.get("size", '/run/poolparams/createsize/')
-
-            # initialize a python pool object then create the underlying
-            # daos storage
-            pool = DaosPool(self.context)
-            pool.create(createmode, createuid, creategid,
-                        createsize, createsetid, None)
-
-            time.sleep(1)
-
-            # okay, get rid of it
-            pool.destroy(1)
-
-        except DaosApiError as excep:
-            print(excep)
-            print(traceback.format_exc())
-            self.fail("6 server test failed.\n")
-
-        except Exception as excep:
-            self.fail("Daos code segfaulted most likely.  Error: %s" % excep)
-
-        # no matter what happens cleanup
-        finally:
-            if self.agent_sessions:
-                agent_utils.stop_agent(self.agent_sessions)
-            server_utils.stop_server(hosts=self.hostlist_servers)
-            os.remove(hostfile_servers)
-
+        counter = 0
+        hostlist_servers = self.hostlist_servers[:6]
+        group_hosts = {self.server_group: hostlist_servers}
+
+        self.start_servers_and_clients(group_hosts)
+        counter = 0
+
+        while counter < 10:
+            self.pool = TestPool(self.context, self.log)
+            self.pool.get_params(self)
+            self.pool.name.value = self.server_group
+            self.pool.create()
+            self.log.info("Pool UUID is %s", self.pool.uuid)
+
+            # Check that the pool was created
+            self.assertTrue(
+                self.pool.check_files(hostlist_servers),
+                "Pool data not detected on servers before destroy")
+
+            # Destroy pool with direct API call
+            try:
+                self.log.info("Attempting to destroy pool")
+                self.pool.pool.destroy(0)
+            except Exception as error:
```

Comment:
(pylint-broad-except) Catching too general exception Exception

Reply:
Changed to:
           try:
                self.log.info("Attempting to destroy pool")
                self.pool.pool.destroy(0)
            except DaosApiError as error: